allow logging in with smartcard and username+password

mrbaseman opened the pull request:
This should perhaps be tested by someone who actively uses smartcard login without username and password. I remember that usual practice was to have dummy entries in username and password which is a configuration which would be broken by this commit. Anyhow, if those users change their config file to include an empty line for username and password, I would expect it to work again, but I would have a better feeling if someone could confirm this.

mrbaseman commented:
@simowdas [has confirmed](https://github.com/adrienverge/openfortivpn/issues/629#issuecomment-610466947) that this works for him and solves #629

mrbaseman commented:
we have the [feature request](https://github.com/adrienverge/openfortivpn/issues/629) for logging in with username+password+smartcard and we have a solution for this in  [my smartcard branch](https://github.com/mrbaseman/openfortivpn/tree/smartcard). I would like to have this change tested in other scenarios (smartcard without username+password or only regular user certificate, i.e. without username+password etc.). 
I went through the older issues and the pull requests in this area and have collected the users who have contributed or reported.
@rmuehl @kamushadenes @claudiuion @nicolaerosia @martinetd @sabino @epsilon77  could you help us and test this change in your environment, please?

If you have a dummy entry for the username in the config file and you log in without username+password, please put empty entries in the config file:
```
username =
password =
```

If you need a password, either put it in the config file or remove the `password` line from the config (or comment it out).

It would be great if you could provide your feedback to this. Thank you.

DimitriPapadopoulos commented:
@mrbaseman Not much feedback about this change. Perhaps we should merge and hope it is tested a little bit more in the master branch.

mrbaseman commented:
Ok. It's a pity that we didn't receive much feedback, even in those times of close down, when VPN solutions have gathered more importance.
As you have suggested, I have rebased, squashed and merged it, in the hope of getting some more testing and perhaps some feedback before the next release.
